File: samples/basic-samples/simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/screens/HomeFragment.java
package com.zhuinden.navigationexamplefrag.screens;

import android.os.Bundle;
import android.view.View;

import com.zhuinden.navigationexamplefrag.R;
import com.zhuinden.simplestack.navigator.Navigator;
import com.zhuinden.simplestackextensions.fragments.KeyedFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by Owner on 2017. 06. 29..
 */

public class HomeFragment
        extends KeyedFragment {
    public HomeFragment() {
        super(R.layout.home_view);
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.home_button).setOnClickListener(v -> {
            Navigator.getBackstack(requireActivity()).goTo(OtherKey.create());
        });

        HomeKey homeKey = getKey(); // get args
    }
}

File: samples/basic-samples/simple-stack-example-basic-kotlin-fragment/src/main/java/com/zhuinden/simplestackexamplekotlinfragment/screens/HomeFragment.kt
package com.zhuinden.simplestackexamplekotlinfragment.screens

import android.os.Bundle
import android.view.View
import com.zhuinden.simplestackexamplekotlinfragment.R
import com.zhuinden.simplestackexamplekotlinfragment.databinding.HomeViewBinding
import com.zhuinden.simplestackexamplekotlinfragment.utils.onClick
import com.zhuinden.simplestackextensions.fragments.KeyedFragment
import com.zhuinden.simplestackextensions.fragmentsktx.backstack

/**
 * Created by Owner on 2017.11.13.
 */

class HomeFragment : KeyedFragment(R.layout.home_view) {
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val binding = HomeViewBinding.bind(view)

        binding.homeButton.onClick {
            backstack.goTo(OtherKey)
        }

        val homeKey = getKey<HomeKey>() // args
    }
}
